include/nbl/asset/ICPUGraphicsPipeline.h

```diff
@@ -29,27 +29,65 @@ class ICPUGraphicsPipeline final : public ICPUPipeline<IGraphicsPipeline<ICPUPip
 					return pipeline_base_t::SCreationParams::impl_valid(std::move(extra));
 				}
 		};
+
 		static core::smart_refctd_ptr<ICPUGraphicsPipeline> create(const SCreationParams& params)
 		{
 			// we'll validate the specialization info later when attempting to set it
-            if (!params.impl_valid([](const IPipelineBase::SShaderSpecInfo& info)->bool{return true;}))
-                return nullptr;
-            auto retval = new ICPUGraphicsPipeline(params);
-            for (const auto spec : params.shaders)
-            if (spec.shader)
-				retval->setSpecInfo(spec);
-            return core::smart_refctd_ptr<ICPUGraphicsPipeline>(retval,core::dont_grab);
+        if (!params.impl_valid([](const SShaderSpecInfo<true>& info)->bool{return true;}))
+            return nullptr;
+        auto retval = new ICPUGraphicsPipeline(params);
+        for (const auto spec : params.shaders)
+        {
+            if (spec.shader) retval->setSpecInfo(spec);
+        }
+        return core::smart_refctd_ptr<ICPUGraphicsPipeline>(retval,core::dont_grab);
 		}
```

Comment:
again, the CPU stuff shouldn't check if stuff is valid at creation time (since its mutable